pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -738,6 +639,46 @@ func (c *Controller) getProxyServiceInstancesByEndpoint(endpoints v1.Endpoints,
 	return out
 }
 
+func (c *Controller) getProxyServiceInstancesByEndpointSlice(slice discoveryv1alpha1.EndpointSlice, proxy *model.Proxy) []*model.ServiceInstance {
+	out := make([]*model.ServiceInstance, 0)
+
+	hostname := kube.ServiceHostname(slice.Labels[discoveryv1alpha1.LabelServiceName], slice.Namespace, c.domainSuffix)
+	c.RLock()
+	svc := c.servicesMap[hostname]
+	c.RUnlock()
+
+	if svc != nil {
+		for _, port := range slice.Ports {
+			if port.Name == nil || port.Port == nil {
+				continue
+			}
+			svcPort, exists := svc.Ports.Get(*port.Name)
+			if !exists {
+				continue
+			}
+
+			podIP := proxy.IPAddresses[0]
+
+			// consider multiple IP scenarios
+			for _, ip := range proxy.IPAddresses {
+				for _, ep := range slice.Endpoints {
+					for _, a := range ep.Addresses {
+						if a == ip {
```

Comment:
Should we skip other endpoints and go to next proxy ip?

Reply:
I am not sure what you mean by this?

Reply:
I was suggesting to break the loop once we get a match.